scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -252,6 +253,24 @@ public boolean isSuccess() {
     }
 
     public boolean isFailure() { return !isSuccess; }
+
+    @Override
+    public int hashCode() {
+      return isSuccess ? value.hashCode() : exception.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj == this) {
+        return true;
+      }
+      if (!(obj instanceof Try)) {
+        return false;
+      }
+      Try<?> that = (Try<?>) obj;
+      return (this.isSuccess == that.isSuccess) && Objects.equals(this.get(), that.get()) &&
+          Objects.equals(this.getException(), that.getException());
```

Comment:
do we need a special `equals()` override for this `Throwable` equality check, too? for example `Objects.equals(new RuntimeException("foo"), new RuntimeException("foo"))` returns false

Reply:
Same behavior with scala `Try` I guess that's acceptable?
This is mainly to avoid tripping mutation detector, but probably not 100% necessary.